### ISSUE TITLE
Watcher stops working when you delete watched files

### ISSUE THREAD

**Issue by jeevatkm** [radovskyb/watcher#22]
@radovskyb Currently watcher stops working when you delete the watched file/directory. It's a bumper for my upcoming release.

It gives an error.
```
2017-07-21 12:01:02.729 ERROR Watch error:error: watched file or folder deleted
```

Doesn't return any events there after. And If you call `w.Close()` at this moment, program gets stuck.

Please let me know if you need further info.

_PS: wether you call `Close` or not, watcher stops working_

**Comment by jeevatkm** [comment]
You can have a look watcher implementation for aah framework. 

https://github.com/go-aah/tools/blob/v0-unstable/aah/run.go#L135-L217

Please let me know, if I'm doing something wrong or way to handle or fix.

**Comment by radovskyb** [comment]
Hey @jeevatkm,

I was going to suggest for you to ignore when that type of error occurs like so:

```go
case err := <-w.Error:
    if err == watcher.ErrWatchedFileDeleted {
	fmt.Println("ignore and continue")
	continue
    }
```

However, I believe there is a deadlock that occurs now when that happens (must have missed it when I rewrote watcher last time), but I will fix it when I get a chance soon!

**Comment by radovskyb** [comment]
Hey @jeevatkm,

I removed the deadlock issue, so you can try again and let me know how it goes :)

**Comment by jeevatkm** [comment]
@radovskyb  Thank you, this is very helpful.

I will very using against master branch and let you know soon. Then you can make a release (tag) for this fix.

**Comment by jeevatkm** [comment]
Working perfectly :) thanks, Can you please make a release?

**Comment by radovskyb** [comment]
Released as v1.0.1 :)
